src/GitExtensions.PluginTemplate/Plugin.cs

```diff
@@ -1,34 +1,55 @@
 ﻿using GitExtensions.PluginTemplate.Properties;
 using GitUIPluginInterfaces;
 using ResourceManager;
-using System;
-using System.Collections.Generic;
 using System.ComponentModel.Composition;
-using System.Diagnostics;
-using System.IO;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Windows.Forms;
 
 namespace GitExtensions.PluginTemplate
 {
     /// <summary>
-    /// An empty Git Extensions plugin implementation.
+    /// A template for Git Extensions plugins.
     /// </summary>
     [Export(typeof(IGitPlugin))]
     public class Plugin : GitPluginBase
     {
         public Plugin()
         {
-            Name = "PluginTemplate";
-            Description = "Plugin Template";
+            /// <summary>
+            /// See: https://github.com/gitextensions/gitextensions.plugintemplate/wiki/GitPluginBase#protected-void-setnameanddescriptionstring-name
+            /// </summary>
+            SetNameAndDescription("Plugin Template");

```

Comment:
This looks like a poor API
It states it sets _name_ and _description_, yet takes a single parameter. It should take two parameters. A description could be optional, set to the name.
The name's length should be limited too (to something sensible, like 50 chars).

Reply:
I did not yet figure out what _description_ was supposed to be used for or has been used for in the past, but I'm pretty sure it is unused right now.
I will dig through the code and, depending on what I find, prepare a PR to clean this part of the API.

@maraf: Do you know something about why there are 2 parameters _name_ and _description_?

Reply:
Unfortunately not. I think `Name` is used as a label on plugins menu item.

Reply:
I figured it out and suggest to resolve this as an own issue: https://github.com/gitextensions/gitextensions.plugintemplate/issues/5